### PR TITLE
first PR

### PR DESCRIPTION
**Pull request by yelizariev**

### CODE REVIEW

File: saas/models/saas_template.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        defalt="[]")
+    template_post_init = fields.Text(
+        'Template Initialization',
+        default=lambda s: s.env['ir.actions.server'].DEFAULT_PYTHON_CODE,
+        help='Python code to be executed once db is created and modules are installed')
+    operator_ids = fields.One2many(
+        'saas.template.operator',
+        'template_id')
+
+    @api.constrains('code')
```
Comment (trojikman) [inline saas/models/saas_template.py]: @yelizariev, shouldn't there be template_post_init field here?

**Review by trojikman — CHANGES_REQUESTED** [review]

File: saas/models/saas_template.py (hunk 1 above)
Comment (yelizariev) [reply saas/models/saas_template.py]: Right. It's supposed to be ``template_post_init`` here :+1:

File: saas/models/saas_template.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                })
+            password = random_password()
+            self.env['saas.log'].log_db_creating(self)
+
+            r.write({
+                'state': 'creating',
+                'password': password,
+            })
+
+            r.db_id.with_delay().create_db(
```
Comment (trojikman) [inline saas/models/saas_template.py]: AttributeError: 'saas.db' object has no attribute 'with_delay'

**Review by trojikman — CHANGES_REQUESTED** [review]

File: saas/models/saas_template.py (hunk 2 above)
Comment (yelizariev) [reply saas/models/saas_template.py]: Try to add ``@job`` decorator

Comment (yelizariev) [reply saas/models/saas_template.py]: Actually, it's there. Check OCA/queue for configuration: https://github.com/OCA/queue/tree/11.0/queue_job#id13

Try to use this module to confirm configuration: https://github.com/OCA/queue/tree/11.0/base_import_async

For test environment check for documentation about how this module should be run: https://github.com/OCA/queue/tree/11.0/test_queue_job

Comment (trojikman) [reply saas/models/saas_template.py]: base_import_async is uninstallable on odoo 12

File: saas/models/saas_template.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            # delete db is there is one
+            r.operator_db_id.drop_db()
+            if not r.operator_db_id or r.operator_id != r.operator_db_id.operator_id:
+                r.operator_db_id = self.env['saas.db'].create({
+                    'name': r.operator_db_name,
+                    'operator_id': r.operator_id.id,
+                    'type': 'template',
+                })
+            password = random_password()
+            self.env['saas.log'].log_db_creating(self)
```
Comment (trojikman) [inline saas/models/saas_template.py]: It is strange that this piece of code just now started to generate an error. The log_db_creating takes instance of saas.db

**Review by trojikman — CHANGES_REQUESTED** [review]

File: saas/models/saas_template.py (hunk 3 above)
Comment (yelizariev) [reply saas/models/saas_template.py]: Can you fix the arguments or the method?

Comment (trojikman) [reply saas/models/saas_template.py]: I already fixed it. I'll post changes soon

**Pull request merged**